src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1264,6 +1262,7 @@ public Status do_POST_Document_revs_diff(Database _db, String _docID, String _at
 
         // Look them up, removing the existing ones from revs:
         try {
+            // query db only, not necessary to be syncrhonized
```

Comment:
In theory, SQLiteDatabase is not (always) application logic thread-safe. SQLite is compiled with multi-threading mode, and Android SQLite database does have read/write connection separately. It's also related to connection pool size. We could chat more in detail.